src/common/kube-helpers.ts

```diff
@@ -23,14 +24,36 @@ function resolveTilde(filePath: string) {
   return filePath;
 }
 
+function readResolvedPathSync(filePath: string): string {
+  return fse.readFileSync(path.resolve(resolveTilde(filePath)), "utf8");
+}
+
+function checkRawContext(rawContext: any): boolean {
+  return rawContext.name && rawContext.context?.cluster && rawContext.context?.user;
+}
+
+function loadToOptions(rawYaml: string): any {
+  const obj = yaml.safeLoad(rawYaml);
+
+  if (typeof obj !== "object" || !obj) {
+    throw new TypeError("KubeConfig root entry must be an object");
+  }
+
+  const { clusters: rawClusters, users: rawUsers, contexts: rawContexts, "current-context": currentContext } = obj;
+  const clusters = newClusters(rawClusters);
+  const users = newUsers(rawUsers);
+  const contexts = newContexts(rawContexts?.filter(checkRawContext));
```

Comment:
Seems a bit weird to just apply `checkRawContent`. This PR originally used `validateKubeConfig()` at this stage (before loading the kubeconfig), right?

So if `name`, `cluster` or `user` is missing the context is silently not included. If cluster is set to an unknown value then the context is included and the user is only informed that it is bad when they try to add it by clicking the Add Cluster button. It would be nice to catch all static problems with the config before the context is added to the "Select contexts" list.

Reply:
I think that the filtering is best (though what might be better is having the contexts in the list but disabled with the errors displayed on hover). Though we can do that in a follow up PR.